tests/unit/valgrind.suppressions

```diff
@@ -10,7 +10,46 @@
    fun:pthread_create@@GLIBC_2.2.5
    fun:main
 }
-
+# Append valgrind suppression for ubuntu22
```

Comment:
I don't clearly remember, but I was thinking that this was an issue for Ubuntu24, not Ubuntu22?

Also I view suppression as necessary, but very evil and something that we try to minimize, so I'd vote to include more context about why these were added
- what tests will fail if these aren't added?
- context that this is the same as the pthread suppressions above, but for a different libc
- maybe link into some tracking issue somewhere, that once we move off of ubuntu18/22, then we can remove the above suppression.

Reply:
This suppression is needed to both Ubuntu22 and Ubuntu24. I will update the information.

I will also provide necessary contexts to those questions in the Valgrind Suppressions to explain. And I will create a tracking number for that.

Reply:
The following tests will failed if that suppression is not included. All of them are Pthread memory defects which prevents the test from exit gracefully.

```
agent-1  | The following tests FAILED:
agent-1  |       82 - s2n_examples_test (Failed)
agent-1  |       95 - s2n_fork_generation_number_test (Failed)
agent-1  |      113 - s2n_init_test (Failed)
agent-1  |      121 - s2n_key_update_threads_test (Failed)
agent-1  |      158 - s2n_random_test (Failed)
agent-1  | -- Processing memory checking output:
agent-1  | 79/268 MemCheck: #82: s2n_examples_test ................................   Defects: 6
agent-1  | 85/268 MemCheck: #95: s2n_fork_generation_number_test ..................   Defects: 6
agent-1  | 102/268 MemCheck: #113: s2n_init_test ....................................   Defects: 1
agent-1  | 116/268 MemCheck: #121: s2n_key_update_threads_test ......................   Defects: 5
agent-1  | 144/268 MemCheck: #158: s2n_random_test ..................................   Defects: 4
```

I have opened a tracking issue #4777 for this comments. I will put that issue number in the comments as well.